util/libmeta/get_signers.go

```diff
@@ -11,6 +11,7 @@ type Metadata interface {
 
 type MsgWithMetadata[T Metadata] interface {
 	GetMetadata() T
+	GetSigners() []string
```

Comment:
Remove this. Use GetMetadata().GetSigners()

Reply:
Reverted back to the old [commit](https://github.com/palomachain/paloma/pull/1060#discussion_r1431172964) `GetMetadata()`,

Reply:
`msg.GetSigners()` used to exist in sdk.Tx `GetMsgs()` in earlier version, but after upgrade its moved, so we updated accordingly, later you have asked to use `m.getSigners` [here](https://github.com/palomachain/paloma/pull/1060#discussion_r1431172964) again we are back to use `msg.GetSigners`  [here](https://github.com/palomachain/paloma/pull/1060#discussion_r1437752513). Does it compile with `msg.GetSigners` as such method does not exist in SDK anymore?

Reply:
We've had an extensive discussion about this on another PR before I believe.

You're correct that `GetSigners()` and `ValidateBasic()` are interfaces that were to be implemented by every message in `v0.47`, and this necessity has been removed with `0.50`, which is why you thought to remove it.

The issue is that Paloma has built a custom ante handler that relies on this interface. The ante handler will skip any messages that do not implement it, which is fine for all message types coming from Cosmos directly.

However, custom messages defined by Paloma still need to implement this interface so the ante handler picks them up accordingly.

This means:

- all custom messages need to explicitly implement `GetSigners()` and `ValidateBasic()`
- every implementation should work identically, using `libmeta`, and not access any other fields directly

There's more background in the comments on this PR: https://github.com/palomachain/paloma/pull/1038